tinyagentos/worker/agent.py

```
@@ -793,6 +848,29 @@ async def run(self):
+                            from tinyagentos.worker.self_update import post_update_startup
+                            outcome = await post_update_startup(
+                                controller_url=self.controller_url,
+                                agent=self,
+                                state_dir=self._state_dir,
+                            )
+                            if outcome is not None:
+                                logger.info(
+                                    "self-update: post-restart outcome=%s",
+                                    outcome.get("outcome", "unknown"),
+                                )
+                        except Exception:
```

Ruff (0.15.21)

[warning] Do not catch blind exception: `Exception`

(BLE001)